flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1713,7 +1713,23 @@ public StreamGraph getStreamGraph() {
 	 */
 	@Internal
 	public StreamGraph getStreamGraph(String jobName) {
-		return getStreamGraphGenerator().setJobName(jobName).generate();
+		return getStreamGraph(jobName, true);
+	}
+
+	/**
+	 * Getter of the {@link org.apache.flink.streaming.api.graph.StreamGraph} of the streaming job.
+	 *
+	 * @param jobName Desired name of the job
+	 * @param clearTransformations Whether or not to start a new stage of execution
+	 * @return The streamgraph representing the transformations
+	 */
+	@Internal
+	public StreamGraph getStreamGraph(String jobName, boolean clearTransformations) {
```

Comment:
In which case flink need to keep the transformation instead of clearing it ?

Reply:
I only saw the usage in Tests, like `DataStreamTest`.

Reply:
When getting the execution graph as a string from the environment, the transformations should not be cleared, for example.

Reply:
So when `getExecutionPlan()` is called the transformations should not be cleared. And if this slipped then we should add a test for that!

Reply:
The `getExecutionPlan()` invoke `getStreamGraph()` directly, if we found that for most of the user cases, we do not need to keep those transformations, i'm inclined to remove the transformations also for `getExecutionPlan()`, this would make these `getXXX` interfaces synced with removing transformations logic. Also make it not that confusing.

Reply:
I do not think that there is any way to know how users use this method, so I would lean towards keeping the existing semantics, rather than changing them.

Reply:
Thanks, i agree.